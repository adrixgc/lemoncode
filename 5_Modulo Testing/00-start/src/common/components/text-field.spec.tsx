import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import { TextField } from './text-field';

afterEach(cleanup);

describe('text-field specs', () => {
    it('should render text-field with label', () => {

        // Arrange
        const props = {
            name: 'text-field_name',
            label: 'Foo',
            value: 'Bar',
            onChange: jest.fn(),
        };

        // Act
        const { getByText } = render(<TextField {...props} />);
        const labelElement = getByText(props.label);

        // Assert
        expect(labelElement).not.toBeUndefined();
    });

    it('should render text-field using snapshot testing', () => {
        // Arrange
        const props = {
            name: 'text-field_name',
            label: 'Foo',
            value: 'Bar',
            onChange: jest.fn(),
        }

        // Act
        const { asFragment } = render(<TextField {...props} />);
        // Assert
        expect(asFragment()).toMatchSnapshot();
    });
});