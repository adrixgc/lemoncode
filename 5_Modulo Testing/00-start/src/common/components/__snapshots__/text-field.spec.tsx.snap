// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-field specs should render text-field using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Foo
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value="Bar"
      />
    </div>
  </div>
</DocumentFragment>
`;
